### ISSUE TITLE
Improve keyboard interaction for accessibility

### ISSUE THREAD

**Issue by joshsloat** [rpearce/react-medium-image-zoom#70]
Per accessibility standards [(WAI-ARIA](https://www.w3.org/TR/wai-aria-practices/#button), [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets), [WebAIM](https://webaim.org/techniques/keyboard/)), it's important to ensure that all content can be accessed with the keyboard alone - and that the keyboard and mouse produce the same experience. 

Since this component supports `click` to zoom, it should also support keyboard interaction to initiate and exit zoom.

- [x] By default, users can only navigate to links, buttons, and form controls with a keyboard. These natively-accessible elements should be used whenever possible. If this is not possible (like this case), you may need to use `tabindex="0"` to ensure an element can receive keyboard focus.
- [x] Following ARIA Practices, the interaction most closely models a button, so it should honor both `Enter` and `Space`. Use `onkeydown` to trap rather than `onkeypress` since IE will not fire keypress events for non-alphanumeric keys. 
- [x] Since the zoomed appearance functions like a modal dialog, focus should return to the image element that opened the dialog.
- [x] Since the zoomed appearance functions like a modal dialog, it should support `Esc` to close as it already does.

**Comment by rpearce** [comment]
Closed via https://github.com/rpearce/react-medium-image-zoom/pull/71 and https://github.com/rpearce/react-medium-image-zoom/pull/73
